WMF Framework/EventPlatformClientLibrary.swift

```diff
@@ -389,63 +386,71 @@ public class EPC: NSObject {
 
-        // Make them available to any newly logged events before flushing
-        // buffer (this is set using serial queue but asynchronously)
-        self.streamConfigurations = streamConfigs
-
-        // Process event buffer after making stream configs available
-        // NOTE: If any event is re-submitted while streamConfigurations
-        // is still being set (asynchronously), they will just go back to
-        // input buffer.
-        var event: (stream: String, data: [String: NSCoding], domain: String?)?
-        while !self.inputBufferIsEmpty() {
-            event = self.inputBufferPopFirst()
-            if let event = event {
-                self.submit(stream: event.stream, data: event.data, domain: event.domain)
+            // Make them available to any newly logged events before flushing
+            // buffer (this is set using serial queue but asynchronously)
+            streamConfigurations = json.streams.reduce(into: [:], { (result, kv) in
+                guard let stream = Stream(rawValue: kv.key) else {
+                    return
+                }
+                result?[stream] = kv.value
+            })
+
+            // Process event buffer after making stream configs available
+            // NOTE: If any event is re-submitted while streamConfigurations
+            // is still being set (asynchronously), they will just go back to
+            // input buffer.
+            while let event = inputBufferPopFirst() {
+                guard inSample(stream: event.stream) else {
+                    continue
+                }
+                appendPostToOutputBuffer((url: streamIntakeServiceURI, body: data))
             }
+        } catch let error {
+            DDLogError("EPC: Problem processing JSON payload from response: \(error)")
         }
     }
 
     /**
      * Flush the queue of outgoing requests in a first-in-first-out,
      * fire-and-forget fashion
      */
-    private func postAllScheduled() {
-        if self.outputBufferIsEmpty() {
-            return
-        }
+    private func postAllScheduled(_ completion: (() -> Void)? = nil) {
         DDLogDebug("EPC: Posting all scheduled requests")
-        var item: (url: URL, body: NSDictionary)?
-        while !self.outputBufferIsEmpty() {
-            item = self.outputBufferPopFirst()
-            if let item = item {
-                self.httpPost(url: item.url, body: item.body)
+        let group = DispatchGroup()
+        while let item = outputBufferPopFirst() {
+            group.enter()
+            httpPost(url: item.url, body: item.body) { result in
+                switch result {
+                case .success:
+                    break
+                case .failure(let error):
+                    switch error {
+                    case .networkingLibraryError:
+                        /// Retry on networking library failure
+                        self.appendPostToOutputBuffer(item)
```

Comment:
Seems like this could get stuck in a retry loop, maybe we should retry up to a max number of times it fails with this error?

Reply:
Good catch - fixed. I added a struct for persisting event requests. This would have prevented the `event.data` bug below as well.

Reply:
Thanks! The request struct feels much better.